lib/hanami/utils/kernel.rb

```diff
@@ -404,15 +413,14 @@ def self.Integer(arg)
       #   # Missing #respond_to?
       #   input = BasicObject.new
       #   Hanami::Utils::Kernel.BigDecimal(input) # => TypeError
-      def self.BigDecimal(arg)
+      def self.BigDecimal(arg, precision: BIGDECIMAL_DEFAULT_PRECISION)
         case arg
+        when ->(a) { a.respond_to?(:to_d) && a.class == Rational } then arg.to_d(precision)
```

Comment:
@AlfonsoUceda Sure, but because `Rational` in case statement appears before of the check for `#to_d`, it will never be hit, unless `arg` is an object that isn't a `Float`, or a `Complex` or a `Rational`.